src/index.ts

```diff
@@ -123,6 +123,13 @@ function fileToText(file: File): Promise<string> {
   });
 }
 
+function extract<T>(promise: AxiosPromise<T>): Promise<T> {
+  return new Promise((resolve, reject) => {
+    promise.then((resp) => resolve(resp.data))
```

Comment:
I didn't think `resp.data` was a promise, it is already populated when the response resolves, in which case wouldn't this always resolve immediately always? The `fetch` API requires another layer of promise resolution (e.g. to get the `json()` of it) but I don't think it is needed for axios. In which case, is `extract` even helpful?

Reply:
On another look, maybe I'm misunderstanding how this function works. You can ignore if so :)

Reply:
An `AxiosPromise<T>` is defined to be `Promise<AxiosResponse<T>>`, which in turn is an axios response whose `data` field contains a value of type `T`. So this is essentially taking `promise`, resolving it, then pulling out the `data` field (which has the actual API response data in it) and packing that in another promise, which can then be returned from any of the async methods in the `MultinetAPI` object.

It's essentially a bit of "semantic sugar" to turn an axios response promise into a Multinet data promise. (Hopefully that explanation made sense!)

Reply:
So then is it true that

```
const data = await something();
```

and

```
const { data } = await _something();
```

are identical? Probably more work than I'd do add that little of syntactic sugar, but this is clearly your call, I'm just swooping in and will just as soon swoop out, no need to please me here :)

Reply:
You have a point, but more likely you'd want to write things like

```javascript
const thing = (await _something()).data;
```

so everything doesn't have to be called `data`, when it's more "ergonomic" to just be able to do:

```javascript
const thing = await something();
```

and fall back on `_something()` when you need to engage more deeply with Axios. Since the raison d'etre for multinetjs is to provide this kind of client app writer comfort, I'm ok with doing the extra work here (especially since it's essentially a constant overhead per function).

If there's a good way to autogenerate the non-underscored versions of these methods, I would use that but for now I just swept through and did the manual work 🤷🏾‍♂️

Reply:
Yes, good point also that this is a library for others to use, not just internal logic, so library user ergonomics can be given more consideration. Carry on.

Reply:
(As a matter of github usage policy, I always ask thread openers to resolve their own threads when satisfied with a discussion, so please feel free to do so if and when you are ready 🙂)